File: office-space-backend/src/main/java/com/ourfancyteamname/officespace/services/ProductService.java
package com.ourfancyteamname.officespace.services;

import org.springframework.data.domain.Page;

import com.ourfancyteamname.officespace.db.entities.Product;
import com.ourfancyteamname.officespace.dtos.ProductDto;
import com.ourfancyteamname.officespace.dtos.TableSearchRequest;

public interface ProductService {

  Page<ProductDto> findByPaging(TableSearchRequest tableSearchRequest);

  Page<ProductDto> findProductWithDisplayName(TableSearchRequest tableSearchRequest);

  Product create(ProductDto productDto);

  Product update(ProductDto productDto);

  void delete(Integer productId);
}

File: office-space-frontend/src/main/web/office-space-web/src/app/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TableSearchRequest } from '../dtos/tableSearch';
import { StorageService } from './auth/storage.service';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  readonly url = `${this.storage.get(StorageService.API)}product`;

  constructor(private http: HttpClient, private storage: StorageService) { }

  create(data): Observable<any> {
    return this.http.post(`${this.url}/create`, data);
  }

  findAll(data: TableSearchRequest): Observable<any> {
    return this.http.post(`${this.url}/list`, data);
  }

  delete(id: number): Observable<any> {
    return this.http.delete(`${this.url}/${id}`);
  }

  update(data): Observable<any> {
    return this.http.post(`${this.url}/update`, data);
  }
}
